consensus/state.go

```diff
@@ -652,9 +652,10 @@ func (cs *State) updateToState(state sm.State) {
 		// to be gathered for the first block.
 		// And alternative solution that relies on clocks:
 		// cs.StartTime = state.LastBlockTime.Add(timeoutCommit)
-		cs.StartTime = cs.config.Commit(tmtime.Now())
+		cs.StartTime = cs.config.Commit(tmtime.Now(), time.Millisecond)
 	} else {
-		cs.StartTime = cs.config.Commit(cs.CommitTime)
+		elapsedTime := cs.CommitTime.Sub(cs.StartTime)
+		cs.StartTime = cs.config.Commit(cs.CommitTime, elapsedTime)
```

Comment:
For me, this is more complex than it needs to be. Can't you just have:
```suggestion
		cs.StartTime = cs.config.Commit(cs.StartTime)
```
and
```go
func (cfg *ConsensusConfig) Commit(startTime time.Time) {
    return startTime.Add(cfg.TargetRoundDuration)
}
```
It seems like consensus can handle `StartTime`'s in the past so I don't think you need to worrry about it being less than now. `CommitTime` really has little to do with it. i.e. 
CommitTime + (TargetRoundDuration - (CommitTime-StartTime) = StartTime + TargetRoundDuration.

Reply:
I'd actually rename `Commit(...)` to `NextStartTime(...)`. It seems more appropriate

Reply:
good idea, renamed